engine/src/mac-window.mm

```diff
@@ -2246,20 +2251,29 @@ void MCMacPlatformHandleDrawSync(NSWindow *window)
 		// The timeout value of 0.02ms is specified to avoid hitting the 60hz redraw limit.
 		if (!s_inside_focus_event && !s_showing_sheet && ![m_delegate inUserReshape])
 		{
-			m_waiting_for_draw = true;
+			NSMutableArray *t_popped_events = [NSMutableArray array];
```

Comment:
Explain what we are doing here with the array in a comment.

Also, use explicit management rather than relying on the auto-release pool as this will be called many times potential without a pool being freed:
```
[[NSMutableArray alloc] init];
```
Then explicitly release later on.

Reply:
Ok, I've changed this to explicit reference management now :)